### ISSUE TITLE
Rotating secret failing - Resource 'my-internal-secret' is in meltdown, requeue reconcile after 30s

### ISSUE THREAD

**Issue by drnic** [cloudfoundry-incubator/cf-operator#800]
Sample YAML files are in https://gist.github.com/drnic/96a4160df8da19c809aea0d328aeb81d

```
$ kaf generate-secret.yaml
$ k get secret my-internal-secret -n kubecf -ojsonpath='{.data.password}' | base64 --decode
CDVIqCF7LY6dLyEGq10BrzaZfRWGDl8dxSV4vuVz8eDGga8AoP84SEc22Ben25mM
```

Now rotate secret:

```
$ kaf rotate-my-internal-secret.yaml
```

But the secret doesn't change

```
$ k get secret my-internal-secret -n kubecf -ojsonpath='{.data.password}' | base64 --decode
CDVIqCF7LY6dLyEGq10BrzaZfRWGDl8dxSV4vuVz8eDGga8AoP84SEc22Ben25mM
```

The rotation logs from cf-operator show an error:

```
cf-operator-9cc8f98dd-kxglp cf-operator 2020-01-31T00:33:43.011Z	DEBUG	boshdeployment-reconciler	reference/reconciles.go:88	Listing BOSHDeployment in namespace 'kubecf' for 'rotate-my-internal-secret'
cf-operator-9cc8f98dd-kxglp cf-operator 2020-01-31T00:33:43.011Z	DEBUG	secret-rotation-reconciler	quarkssecret/secret_rotation_controller.go:43Create predicate passed for 'rotate-my-internal-secret'
cf-operator-9cc8f98dd-kxglp cf-operator 2020-01-31T00:33:43.011Z	INFO	secret-rotation-reconciler	quarkssecret/secret_rotation_reconciler.go:50Reconciling QuarksSecret rotation kubecf/rotate-my-internal-secret
cf-operator-9cc8f98dd-kxglp cf-operator 2020-01-31T00:33:43.011Z	DEBUG	quarks-statefulset-reconciler	reference/reconciles.go:88	Listing QuarksStatefulSet in namespace 'kubecf' for 'rotate-my-internal-secret'
cf-operator-9cc8f98dd-kxglp cf-operator 2020-01-31T00:33:43.013Z	DEBUG	controller-runtime.manager.events	recorder/recorder.go:52	Normal	{"object": {"kind":"ConfigMap","namespace":"kubecf","name":"rotate-my-internal-secret","uid":"df2651fb-36d0-461b-935c-eba73d95f619","apiVersion":"v1","resourceVersion":"36898"}, "reason": "Predicates", "message": "{\"reconciliationObjectName\":\"rotate-my-internal-secret\",\"reconciliationObjectKind\":\"corev1.ConfigMap\",\"predicateObjectName\":\"rotate-my-internal-secret\",\"predicateObjectKind\":\"corev1.ConfigMap\",\"namespace\":\"kubecf\",\"message\":\"Create predicate passed for 'rotate-my-internal-secret'\",\"type\":\"Predicates\"}"}
cf-operator-9cc8f98dd-kxglp cf-operator 2020-01-31T00:33:43.027Z	DEBUG	controller-runtime.controller	controller/controller.go:242	Successfully Reconciled	{"controller": "secret-rotation-controller", "request": "kubecf/rotate-my-internal-secret"}
cf-operator-9cc8f98dd-kxglp cf-operator 2020-01-31T00:33:43.027Z	DEBUG	quarks-secret-reconciler	quarkssecret/quarkssecret_controller.go:65	Update predicate passed for 'my-internal-secret'
cf-operator-9cc8f98dd-kxglp cf-operator 2020-01-31T00:33:43.027Z	INFO	quarks-secret-reconciler	quarkssecret/quarkssecret_reconciler.go:86	Reconciling QuarksSecret kubecf/my-internal-secret
cf-operator-9cc8f98dd-kxglp cf-operator 2020-01-31T00:33:43.027Z	DEBUG	quarks-secret-reconciler	controller/controller.go:216	Resource 'my-internal-secret' is in meltdown, requeue reconcile after 30s
cf-operator-9cc8f98dd-kxglp cf-operator 2020-01-31T00:33:43.027Z	DEBUG	controller-runtime.manager.events	recorder/recorder.go:52	Normal	{"object": {"kind":"QuarksSecret","namespace":"kubecf","name":"my-internal-secret","uid":"78df9c53-50cf-4ad9-b256-8a533d9f9f7b","apiVersion":"quarks.cloudfoundry.org/v1alpha1","resourceVersion":"36900"}, "reason": "Predicates", "message": "{\"reconciliationObjectName\":\"my-internal-secret\",\"reconciliationObjectKind\":\"qsv1a1.QuarksSecret\",\"predicateObjectName\":\"my-internal-secret\",\"predicateObjectKind\":\"qsv1a1.QuarksSecret\",\"namespace\":\"kubecf\",\"message\":\"Update predicate passed for 'my-internal-secret'\",\"type\":\"Predicates\"}"}
cf-operator-9cc8f98dd-kxglp cf-operator 2020-01-31T00:33:43.027Z	DEBUG	controller-runtime.manager.events	recorder/recorder.go:52	Normal	{"object": {"kind":"QuarksSecret","namespace":"kubecf","name":"my-internal-secret","uid":"78df9c53-50cf-4ad9-b256-8a533d9f9f7b","apiVersion":"quarks.cloudfoundry.org/v1alpha1","resourceVersion":"36900"}, "reason": "Meltdown", "message": "Resource 'my-internal-secret' is in meltdown, requeue reconcile after 30s"}
```

**Comment by drnic** [comment]
Deleting the cf-operator pod resolves the issue just once -- the rotation is reconciled once.

**Comment by andreas-kupries** [comment]
This may be PEBKAC ... @fargozhu noted in https://github.com/SUSE/kubecf/issues/44 that the command to apply the rotation request missed the `-n kubecf` to place the request into the watched namespace.

Adding this option and trying again I see the operator properly reacting to the request, and updating the secret in question.

Currently updating my docs.

**Comment by rohitsakala** [comment]
Hello @drnic ,

I tried to reproduce your error, but failed. For me, the secret rotation works. These are the steps I followed.

1. Deployed cf-operator into cfo namespace using helm.
2. kubectl -n cfo create generate-secret.yaml
3. Get secret - `jCSPKAM74KNlMnEWX9GOyquWgPFmgoEoPipD6w6UKFqfNSBVXmHIGHNYF1Owl4La`
4. kubectl -n cfo apply rotate-my-internal-secret.yaml
5. Get secret - `tJA8D78yplegrJpgghjSPuMOvdnuy4D4TqGlMkAUehQnbRTRpPPvEwRpVXC5vn6f`

I am using the latest master of `cf-operator`.
